src/script/solver.h

```diff
@@ -22,6 +22,7 @@ template <typename C> class Span;
 enum class TxoutType {
     NONSTANDARD,
     // 'standard' transaction types:
+    ANCHOR, //!< anyone can spend script
```

Comment:
nit:  If I'm reading the history correctly, there was some initial conversation of a non-segwit ephemeral anchor (e.g. just `OP_TRUE` (and with no empty scriptsig guarantees that segwit offers), for situations not caring much for malleability resistance). This segwit anchor seems like a much better way to go, but I'm not sure if other anchor types are still being considered (and nothing sticks out particularly as a reason why we would even really need more than one type).  

Would this be the only anticipated anchor type to be introduced?  If so, `ANCHOR` seems reasonable.  If not, then the enum (and locations using it) could use a different name to be more specific.